util/parsenum.h

```diff
@@ -64,7 +64,9 @@ _Pragma("clang diagnostic pop")
 			((*(x)) = parsenum_float((s), (min), (max))) :	\
 		(((*(x)) = -1) <= 0) ?					\
 			((*(x)) = parsenum_signed((s), (min), (max))) :	\
-			(((*(x)) = parsenum_unsigned((s),		\
+			(assert(((min) >= 0) && "PARSENUM unsigned with negative minimum bound"),			\
+			assert(((max) <= *(x)) && "PARSENUM unsigned upper bound does not fit in datatype"),			\
```

Comment:
when you wrote
```
TEST(&s, "1234", -123, 4000);
```
was it deliberate?
(BTW, it took me a few minutes before I trained myself to read `s` as `size_t` instead of as `signed`)